### DISCUSSION TITLE
Should we move the `cancelMultiple` function to the periphery?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#264] [Ideas]
We are struggling to keep the size of the `SablierV2Pro` contract under 24kB. It is currently at ~22.8kB, and I fear that after we implement #239, #246, and #255, we will go over the limit.

If I now think about it, I can't remember why we implemented the `cancelAll` function in the core contract. Was there a real reason, or was the motive a bit of laziness to move them to a different repo?

I realize why we did it for `withdrawMultiple` - this is a function that is mostly useful for the recipient. But the converse applies to `cancelAll` - it's a function mostly useful for the sender. Therefore, just like `createMultiple` (that batch-creates streams), we could easily move this function to the periphery.

Any thoughts, @andreivladbrg and @razgraf?

**Comment by razgraf** [comment]
The `cancel` method is `onlySender` right? I guess you won't be able to perform it through a proxy if it checks `msg.sender === stream.sender`. Hence, it having to live in the core contract.

**Reply by PaulRBerg** [reply]
`cancel` is `onlySenderOrRecipient`.

Anyway, this is a great point, which instantly invalidates my idea. Thanks, Razvan.

**Reply by PaulRBerg** [reply]
We will need to figure out other ways to lower the size of `SablierV2Pro` (saving my ideas for another discussion).

**Comment by PaulRBerg** [comment]
Locking based on @razgraf's observation above, and the fact that the way to go regarding the size of the pro contract is [to lower the number of optimizer runs](https://github.com/sablierhq/v2-core/discussions/279).
